src/python/pants/goal/run_tracker.py

```diff
@@ -304,8 +304,9 @@ def store_stats(self):
     # Upload to remote stats db.
     stats_url = self.get_options().stats_upload_url
     if stats_url:
-      t = threading.Thread(target=self.post_stats, args=(stats_url, stats, self.get_options().stats_upload_timeout))
-      t.start()
+      pid = os.fork()
+      if pid == 0:
+        self.post_stats(stats_url, stats, timeout=self.get_options().stats_upload_timeout)
```

Comment:
`post_stat` already handles the exception internally: https://github.com/wisechengyi/pants/blob/205b852ecb6c452d6874fa24516ae34e94a625a6/src/python/pants/goal/run_tracker.py#L262-L268. Would that still be necessary?